app/middleware/authenticator.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require './app/models/user'
+
+def logged_in?
+  !session[:user_id].nil?
+end
+
+# Define the condition
+set(:auth) do |_condition|
+  condition do
+    if logged_in?
+      @user = User.find_by_id(session[:user_id])
+      @admin = @user.admin?
```

Comment:
`@is_admin` is more relevant than `@admin` because the right hand side of the equition is binary